### ISSUE TITLE
React context (and cold context) and Meteor reactive data 

### ISSUE THREAD

**Issue by rclai** [meteor/react-packages#66]
Are you guys going to add support for using React's child context and context to use Meteor reactive data sources? I know they haven't ironed that out totally yet but one of the things that concerns me if I use React is that due to their top down approach, if I want to access a parent component prop from a grand grand grand grand  child, I have to pass it all the way down from the top parent until it reaches that child. How have you guys dealt with this?

**Comment by stubailo** [comment]
There's nothing to stop you using the React context as you see fit!

You can see how in the todos example app, we are using it to open/close the side menu by passing a function down through context: https://github.com/meteor/react-packages/blob/0305155d7b8fb5ab605ddfa375b80ecafa8dbf81/examples/react-todos/client/components/AppBody.jsx#L41-L43

**Comment by rclai** [comment]
Oh okay, but I was specifically asking if you can use a Meteor reactive data source as a child context and whether it's reactivity will be taken into account in the children.

**Comment by rclai** [comment]
Just played around and found out I could just do:

```jsx
...
  mixins: [ReactMeteorData],
  getMeteorData() {
    return {
      tasks: Tasks.find({}, { sort: { createdAt: -1 } }).fetch()
    }
  },
  childContextTypes: {
    test: React.PropTypes.string.isRequired
  },
  getChildContext() {
    return {
      test: (this.data.tasks[0] || {}).text
    }
  },
...
```

**Comment by rclai** [comment]
[Related](https://github.com/meteor/react-packages/issues/19).

**Comment by stubailo** [comment]
Yeah, this looks like a really great way of making the data localized to the subtree, but without all of the boilerplate of passing stuff through `props`.

**Comment by rclai** [comment]
As I dive deeper into using React context, I am finding that we might need to keep an eye on [this issue](https://github.com/facebook/react/issues/2517), especially if you use `shouldComponentUpdate` on middle components.

**Comment by mariorodriguespt** [comment]
https://github.com/meteor/react-packages/pull/197
